embedded-graphics/src/primitives/rounded_rectangle.rs

```diff
@@ -0,0 +1,768 @@
+//! The rounded rectangle primitive.
+
+use crate::{
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    primitives::{
+        ellipse_quadrant::{self, EllipseQuadrant, Quadrant},
+        rectangle::{self, Rectangle},
+        ContainsPoint, Primitive,
+    },
+    style::{PrimitiveStyle, Styled},
+    transform::Transform,
+    DrawTarget,
+};
+
+/// The definition of each corner radius for a rounded rectangle.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct CornerRadii {
+    /// Top left corner radius
+    pub top_left: Size,
+
+    /// Top right corner radius
+    pub top_right: Size,
+
+    /// Bottom right corner radius
+    pub bottom_right: Size,
+
+    /// Bottom left corner radius
+    pub bottom_left: Size,
+}
+
+impl CornerRadii {
+    /// Create a new set of corner radii with all corners having equal values.
+    pub fn new_equal(radius: Size) -> Self {
+        Self {
+            top_left: radius,
+            top_right: radius,
+            bottom_right: radius,
+            bottom_left: radius,
+        }
+    }
+
+    /// Confine corner radii that are too large to a given bounding rectangle
+    fn confine(self, bounding_box: Size) -> Self {
+        // Compute free space for each edge. If there is remaining free space, the call to
+        // checked_sub() will return `None`.
+        let free_space_top =
+            (self.top_left.width + self.top_right.width).checked_sub(bounding_box.width);
```

Comment:
Would `saturating_sub` and using `largest_overlap > 0` as the condition below also work? This way the largest overlap could be calculated without having to deal with the `Option`s.
```suggestion
            (self.top_left.width + self.top_right.width).saturating_sub(bounding_box.width);
```